### DISCUSSION TITLE
Multiple outputs

### DISCUSSION THREAD

**Discussion by vasicstefan94** [Ironclad/rivet#242] [General]
Hi!
I'm wondering if its possible to stream the outputs. What I mean is; I have a chain of text -> chat -> output -> text -> chat -> output etc and it would be very helpful to get the individual outputs as soon as they are done so that I do not have to wait for everything to finish until the runGraphInFile completes. 

Thanks in advance!

**Comment by kitckso** [comment]
The API has streamNode() function to stream a specific node output by name or id.

**Comment by vasicstefan94** [comment]
And how should that be used from the rivet-node npm package? I believe it should be through the createProcessor somehow but the docs are empty:
https://rivet.ironcladapp.com/docs/api-reference/node/createProcessor. 
My goal is to continue running graphs using the runGraphInFile function and get the output from the nodes streamed to my server. 

Thanks in advance!

**Reply by abrenneke** [reply]
Yeah the above answer is correct. Sorry that it's not documented yet. You do have to use `createProcessor` and not `runGraphInFile` when you're doing more advanced stuff like this. The general approach is:

```ts
const { run, streamNode } = createProcessor({ ...same options as runGraphInFile });

const nodeStream = streamNode(nodeIdOrTitle);

await run();
```

The `nodeStream` is a Server-Sent-Events stream that returns `data: { etc }`, so you can do with that whatever you like. If you don't want an SSE stream you can do `const { run, getEvents } = createProcessor({ etc })`, and `getEvents` returns an async generator you can consume while `run` is running.
